core/src/main/java/kafka/server/share/SharePartitionManager.java

```diff
@@ -598,6 +632,12 @@ void processShareFetch(ShareFetch shareFetch) {
             sharePartitions.put(topicIdPartition, sharePartition);
         }
 
+        // Update the metrics for the topics for which we have received a share fetch request.
+        topics.forEach(topic -> {
+            brokerTopicStats.allTopicsStats().totalShareFetchRequestRate().mark();
```

Comment:
We already have a generic metric that tracks the rate for each type of request. So, we probably don't need this one.

Reply:
But shouldn't we have something similar like `totalShareFetchRate` for share fetch as it's tracking per topic?

Reply:
We can keep the per topic one, but we don't need the one that aggregates across all topics.

Reply:
I might be missing something here but this metrics is similar to `TotalFetchRequestsPerSec` as it gives the aggregated view of the `all` topic metrics. Are you suggesting that there should be a cleanup for existing metrics as well to not emit the aggregated ones in future? Sorry I think I am missing something basic here so thought to clarify prior making further change.
https://github.com/apache/kafka/blob/837fb1ed0250837c2d9578ce0e290abded75bcf8/storage/src/main/java/org/apache/kafka/storage/log/metrics/BrokerTopicMetrics.java#L43

Reply:
TotalFetchRequestsPerSec is also duplicating the generic request rate metric. Since it's already there, we can clean it up later. However, it would be useful to avoid introducing duplicated metrics for the new code.

Reply:
@chia7712 yes you are right, as discussed in the thread as well [here](https://github.com/apache/kafka/pull/18444#discussion_r1909573910), I have created the task to fix that: https://issues.apache.org/jira/browse/KAFKA-18640